File: fakeNews.js
var fakeNews = (str) => {
    var fake = 'heidi';
    var index = 0;

    for (var i = 0; i < str.length; i++) {
        if (str[i] == fake[index]) index++;
        if (index == fake.length) return 'YES';
    }
    return 'NO';
}

console.log(fakeNews('abcheaibcdi')); //YES
console.log(fakeNews('hiedi')); //NO

File: fakeNews.cpp
#include <iostream>
using namespace std;

int main(){

    string s;
    cin >> s;

    string standard = "heidi";
    int index  = 0;

    for(int i=0;i< s.length();i++){
        if(s.at(i)==standard.at(index)) index ++;
        if(index ==standard.length()) {
            cout <<"YES";
            return 0;
        }
    }
    
    cout <<"NO";
    return 0;
}